force-app/main/default/aura/RD2_EntryForm/RD2_EntryFormHelper.js

```diff
@@ -23,7 +23,7 @@
      * @description: Redirect the page to either parent or RD record
      */
     redirectToPage: function (component, event) {
-        let navigateToId = event.getParams('detail').recordId;
+        let navigateToId = event.getParams('detail').recordId || component.get('v.parentId');
```

Comment:
Flipping this logic will allow us to redirect back to the parent when there is no `recordId` such as when the user clicks the `cancel` button